engine/tests/kernel_tests/window_overlap_kernel_test.cpp

```diff
@@ -1293,33 +1409,417 @@ TEST_F(WindowOverlapAccumulatorTest, BigWindowSingleNode) {
+
+    CudfTableView full_data_cudf_view ({col0, col1, col2});
+
+    int preceding_value = 50;
+    int following_value = 10;
+    std::vector<cudf::size_type> batch_sizes = {5000, 50000, 20000, 15000, 10000}; // need to sum up to size
+
+    // define how its broken up into batches and overlaps
+    std::vector<std::string> names({"A", "B", "C"});
+    std::vector<std::unique_ptr<CacheData>> preceding_overlaps, batches, following_overlaps;
+    std::vector<std::unique_ptr<CacheData>>& inputCacheData = batches;
+    std::tie(preceding_overlaps, batches, following_overlaps) = break_up_full_data(full_data_cudf_view, preceding_value, following_value, batch_sizes, names);
+
+    std::vector<std::unique_ptr<BlazingTable>> expected_batch_out = make_expected_generator_output(full_data_cudf_view,
+                                                                                                   batch_sizes, names);
+
+    // create and start kernel
+    // Context
+    std::shared_ptr<Context> context = make_context(1);
+
+    auto & communicationData = ral::communication::CommunicationData::getInstance();
+    communicationData.initialize("0", "/tmp");
+
+    std::vector<std::unique_ptr<CacheData>> batches_preceding_pulled,
+                                            batches_pulled,
+                                            batches_following_pulled;
+
+    std::string project_plan = "LogicalProject(min_val=[MIN($0) OVER (ORDER BY $1 ROWS BETWEEN 50 PRECEDING AND 10 FOLLOWING)])";
 
-// FERNANDO TODO
+    std::tie(batches_preceding_pulled,
+             batches_pulled,
+             batches_following_pulled)
+            = run_overlap_generator_kernel(project_plan,
+                                           context,
+                                           inputCacheData);
 
-// create two unit tests for OverlapGeneratorKernel
-// Both can be single node, since this kernel never communicates with other nodes
-// One will have batch sizes and a window size that will generate complete overlaps (imitate BasicSingleNode)
-// the other test will have some batches that are too small and will generate some incomplete overlaps  (imitate BigWindowSingleNode)
+    EXPECT_EQ(batches_preceding_pulled.size(), preceding_overlaps.size());
+    EXPECT_EQ(batches_pulled.size(),           expected_batch_out.size());
+    EXPECT_EQ(batches_following_pulled.size(), following_overlaps.size());
 
-/* Some things to consider when making these tests and the kernels:
-OverlapGeneratorKernel has one input and three outputs:
-* - "batches"
-* - "preceding_overlaps"
-* - "following_overlaps"
-you will want to validate the metadata for preceding_overlaps, following_overlaps
-all you need to validate is if its complete or incomplete
-ral::cache::OVERLAP_STATUS, DONE_OVERLAP_STATUS  or ral::cache::OVERLAP_STATUS, INCOMPLETE_OVERLAP_STATUS
+    for (const auto &batch : batches_preceding_pulled){
+        ral::cache::MetadataDictionary metadata = batch->getMetadata();
```

Comment:
for `batches_preceding_pulled` and `batches_following_pulled` below, you do want to validate the OVERLAP_STATUS the way you are doing here, but you also want to validate the data. 
Use preceding_overlaps,  following_overlaps from `break_up_full_data` like I mention in the comment above

Reply:
Oh, nvm, i see that you do it in a later step